vms/platformvm/warp/payload/README.md

```diff
@@ -0,0 +1,44 @@
+# Payload
+
+An Avalanche Unsigned Warp Message already includes a `networkID`, `sourceChainID`, and `payload` field. The `payload` field is parsed into one of the types included in this package to be further handled by the VM.
+
+## AddressedCall
+
+AddressedCall:
+```
++---------------------+--------+----------------------------------+
+|             codecID : uint16 |                          2 bytes |
++---------------------+--------+----------------------------------+
+|              typeID : uint32 |                          4 bytes |
++---------------------+--------+----------------------------------+
+|       sourceAddress : []byte |                 4 + len(address) |
++---------------------+--------+----------------------------------+
+|             payload : []byte |                 4 + len(payload) |
++---------------------+--------+----------------------------------+
+                               | 14 + len(payload) + len(address) |
+                               +----------------------------------+
+```
+
+- `codecID` is the codec version used to serialize the payload and is hardcoded to `0x0000`
+- `typeID` is the payload type identifier and is `0x00000000` for `AddressedCall`
+- `sourceAddress` is the address that sent this message from the source chain
+- `payload` is an arbitrary byte array payload
+
+## BlockHash
```

Comment:
Should we call this `ProofHash`?

Reply:
As we discussed, this may not be strictly a block hash

Reply:
This is currently used as a `BlockHash` and making it more generic means that the caller would need to know what kind of hash it's referring to (some context is already needed ofc), so I opted to leave this as `BlockHash` for now.

If we want to change the name in the future, it is backwards compatible to do so, so I'd prefer to leave this until we actually use it as something other than a `BlockHash`.

Reply:
As you alluded to, blocks aren't "universal" so the same problem applies to them.

I think locking this terminology in when we communicate it broadly will make us appear a little short-sighted, whereas using the `Proof`/`ProofRoot` moniker is forward-looking.

Reply:
Tbh I'd expect using a `BlockHash` and using that to prove out the rest of the header contents to be simpler.

Do you see this as a more targeted? If a VM would support multiple different types of proofs, how do you foresee them providing context as to what it's proving? Do you foresee them being registered under different typeIDs?

Reply:
I guess I was really excited by the generality/simplicity of using an arbitrary proof structure. For example, it may be way easier to write a solidity contract to verify a trie of warp message IDs rather than a HyperSDK block. That approach also provides a generic interface that any VM can implement and use previously tested/implemented contracts (which I think will be a huge unlock).

We can enshrine "type 0" of the type to be "block hash" if you want to call that out.

Reply:
In terms of using "hash" in the name, I'm not sure what it should be called. I'm not sure all "proof roots" are "hashes" (maybe they are)?

Reply:
Ok I guess the context will be up to the Solidity contracts to know what they are talking to based on the chainID, so that they know how it should be interpreted.

As for naming, I'm fine with `Hash` or `Proof`